curated_transformers/models/attention.py

```diff
@@ -1,10 +1,37 @@
 from typing import Optional
+from dataclasses import dataclass
 import math
 import torch
 from torch import Tensor
 from torch.nn import Module
 
 
+@dataclass
+class AttentionMask:
+    bool_mask: Tensor
+    _logit_mask: Optional[Tensor] = None
+
+    def __post_init__(self):
+        if self.bool_mask.dtype != torch.bool:
+            raise ValueError(
+                f"attention mask of dtype torch.bool expected, was {self.bool_mask.dtype}"
+            )
+
+    @property
+    def logit_mask(self) -> Tensor:
+        if self._logit_mask is None:
+            # The value is `torch.finfo(attn_scores.dype).min`. Unfortunately,
+            # we cannot use `torch.finfo` in TorchScript.
+            self._logit_mask = (1.0 - self.bool_mask.int()) * -3.4028234663852886e38
```

Comment:
Can't we get the minimum value from the built-in `float` type (instead of using the constant directly)?

Reply:
`float` is double precision and we need single precision.